src/reconcile/rec_write.c

```diff
@@ -1997,6 +2002,25 @@ __rec_leaf_page_max(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 	return (page_size * 2);
 }
 
+#define	WT_REC_MAX_SAVED_UPDATES	100
+
+/*
+ * __rec_need_split --
+ *	Check whether adding some bytes to the page requires a split.
+ */
+static bool
+__rec_need_split(WT_RECONCILE *r, size_t next_len)
+{
+	if (r->page->type == WT_PAGE_ROW_LEAF &&
+	    r->supd_next >= WT_REC_MAX_SAVED_UPDATES)
+		return (true);
+
+	if (r->raw_compression)
+		return (next_len > r->space_avail);
+	else
+		return (WT_CHECK_CROSSING_BND(r, next_len));
+}
```

Comment:
I understand what `__rec_need_split()` does, and I think it will work, but nothing in the ticket, commit or code explains why we're doing it?